src/Bridges/bridgeoptimizer.jl

```diff
@@ -109,76 +115,133 @@ function MOI.is_valid(b::AbstractBridgeOptimizer, ci::CI{F, S}) where {F, S}
         return MOI.is_valid(b.model, ci)
     end
 end
+function MOI.is_valid(b::AbstractBridgeOptimizer,
+                      ci::CI{MOI.SingleVariable, S}) where S
+    if is_bridged(b, typeof(ci))
+        return haskey(b.single_variable_constraints, (ci.value, S))
+    else
+        return MOI.is_valid(b.model, ci)
+    end
+end
 function MOI.delete(b::AbstractBridgeOptimizer, vi::VI)
-    for i in eachindex(b.bridges)
-        if b.bridges[i] !== nothing && b.constraint_types[i][1] == MOI.SingleVariable
-            ci = _constraint_index(b, i)
-            f = MOI.get(b, MOI.ConstraintFunction(), ci)::MOI.SingleVariable
-            if f.variable == vi
-                MOI.delete(b, ci)
-            end
+    # Delete all `MOI.SingleVariable` constraint of this variable
+    for key in keys(b.single_variable_constraints)
+        if key[1] == vi.value
+            MOI.delete(b, MOI.ConstraintIndex{MOI.SingleVariable,
+                                              key[2]}(vi.value))
         end
     end
     MOI.delete(b.model, vi)
 end
+function _remove_bridge(b, ci)
+    b.bridges[ci.value] = nothing
+end
+function _remove_bridge(b, ci::MOI.ConstraintIndex{MOI.SingleVariable, S}) where S
+    delete!(b.single_variable_constraints, (ci.value, S))
+end
 function MOI.delete(b::AbstractBridgeOptimizer, ci::CI)
     if is_bridged(b, typeof(ci))
         if !MOI.is_valid(b, ci)
             throw(MOI.InvalidIndex(ci))
         end
         MOI.delete(b, bridge(b, ci))
-        b.bridges[ci.value] = nothing
+        _remove_bridge(b, ci)
     else
         MOI.delete(b.model, ci)
     end
 end
 
 # Attributes
-function MOI.get(b::AbstractBridgeOptimizer,
-                 attr::MOI.ListOfConstraintIndices{F, S}) where {F, S}
+# List of indices of all constraints, even those bridged
+function get_all_even_bridged(
```

Comment:
Why does `_remove_bridged` have an underscore and this function doesn't?

Reply:
I don't want to user to call `_remove_bridged` as it could corrupt the integrity of the object.
`get_all_including_bridged` provides a well defined information which could be useful for inspecting and debugging.